core/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -81,7 +81,8 @@ protected void masterOperation(final ClusterStateRequest request, final ClusterS
         if (request.routingTable()) {
             if (request.indices().length > 0) {
                 RoutingTable.Builder routingTableBuilder = RoutingTable.builder();
-                for (String filteredIndex : request.indices()) {
+                String[] indices = indexNameExpressionResolver.concreteIndexNames(currentState, request);
```

Comment:
This change also affects how the cluster state api works. I guess it means that the cluster state api doesn't resolve indices? I wonder if it's correct to apply the fix here.